src/app/submission/sections/form/section-form.component.ts

```diff
@@ -261,10 +261,10 @@ export class SubmissionSectionFormComponent extends SectionModelComponent {
 
     switch (scope) {
       case SubmissionScopeType.WorkspaceItem: {
-        return this.submissionObject.type === WorkspaceItem.type;
+        return (this.submissionObject as any).type === WorkspaceItem.type.value;
```

Comment:
@FelixNicol 

shouldn't it be more like the following
``` 
(this.submissionObject as any).type.value === WorkspaceItem.type.value
```

or am i missing something ?

Reply:
Hi @atarix83, 

Took me a while to understand the problem. The "type" in the submissionObject is unserialised as a String and not as a ResourceType. That is why the original equality comparison didnt work properly: 
<img width="542" alt="image" src="https://user-images.githubusercontent.com/98552762/193051889-4a59b2eb-da5b-4ac9-a97a-e4ec50ed6b1d.png">
              

If I use type.value on the left like you propose it will not work because we cannot do a .value on a string:
<img width="385" alt="image" src="https://user-images.githubusercontent.com/98552762/193052910-36ef1557-e21f-4ce0-ab7d-285725c2b02b.png">


On the left we have a string and on the right we have a ResourceType. This will therefore always return false. Then if I just write "this.submissionObject.type === WorkspaceItem.type.value" Angular will not allow it, because of course it is the wrong type. 

So I used boxing (as any) and now Angular no longer infer "type" as a ResourceType.

We have similar patches elsewere too. For example; app-routing-paths.ts, remote-data-build.service.ts, delete-comcol-page.component.ts, form-builder.service.ts or section-form-operations.service.ts. 

When I tested the solution, I only tested the case with SubmissionScopeType.WorkflowItem but u guess it should be the same with WorkspaceItem.

Reply:
Here's some more details from the vscode debugger. 

Spy of submissionObject where we see type is a string and not a ResourceType:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/98552762/193084893-4e531544-f10d-46f5-afe5-8fa4f35a38df.png">

I added some line to spy the process:
<img width="568" alt="image" src="https://user-images.githubusercontent.com/98552762/193087702-e8d9e270-5289-4116-b9ba-cc407a5228eb.png">

And here's the result:
<img width="202" alt="image" src="https://user-images.githubusercontent.com/98552762/193087762-e79e33f0-cbad-442e-b87b-9bf4f7401bcf.png">